railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
I see https://github.com/rails/rails/blob/ac027338e4a165273607dccee49a3d38bc836794/actionpack/lib/action_dispatch/routing/route_set.rb#L319-L322 which can be propagated ahead.

I am not 100% sure, we should be this though. Couple of use cases:
- Rails Application with API Engine(right now only way is we explicitly specify this in Engine routes)
- Rails API Application with normal Engine(right now we are creating only 5 routes, using above way we can overcome this)
- Rails API Application with API Engine(Do we have some config user passes, that says even if this is engine, just create 5 routes)

Instead of this, I would prefer we stick to explicit routes.